code/game/mecha/mecha_wreckage.dm

```diff
@@ -103,6 +103,23 @@
 	name = "Marauder wreckage"
 	icon_state = "marauder-broken"
 
+/obj/effect/decal/mecha_wreckage/marauder/New()
+	..()
+	var/list/parts = list(
+		/obj/item/mecha_parts/part/marauder_torso,
+		/obj/item/mecha_parts/part/marauder_head,
+		/obj/item/mecha_parts/part/marauder_left_arm,
+		/obj/item/mecha_parts/part/marauder_right_arm,
+		/obj/item/mecha_parts/part/marauder_left_leg,
+		/obj/item/mecha_parts/part/marauder_right_leg,
+		)
+	for(var/i=0;i<2;i++)
+		if(!isemptylist(parts) && prob(40))
```

Comment:
I love how pessimistic this code is.
"Let's make sure the list I just made with more than one elements has at least one element after I maybe take one away."